lib/mix/lib/mix/project_stack.ex

```diff
@@ -2,188 +2,201 @@ defmodule Mix.ProjectStack do
+      get_and_update(fn %{stack: stack} = state ->
         {top, [mid | bottom]} = Enum.split_while(stack, &(not &1.recursing?))
         {{top, mid.file}, %{state | stack: [%{mid | recursing?: false} | bottom]}}
-      end
+      end)
 
     try do
-      File.cd! Path.dirname(file), fun
+      File.cd!(Path.dirname(file), fun)
     after
-      cast fn %{stack: [mid | bottom]} = state ->
+      cast(fn %{stack: [mid | bottom]} = state ->
         %{state | stack: top ++ [%{mid | recursing?: true} | bottom]}
-      end
+      end)
     end
   end
 
   @spec post_config(config) :: :ok
   def post_config(config) do
-    cast fn state ->
+    cast(fn state ->
       %{state | post_config: Keyword.merge(state.post_config, config)}
-    end
+    end)
   end
 
   @spec printable_app_name() :: atom | nil
   def printable_app_name do
-    get_and_update fn %{stack: stack} = state ->
+    get_and_update(fn %{stack: stack} = state ->
       case stack do
         [] ->
           {nil, state}
+
         [%{io_done: true} | _] ->
           {nil, state}
+
         [h | t] ->
           h = %{h | io_done: true}
           t = Enum.map(t, &%{&1 | io_done: false})
           {h.config[:app], %{state | stack: [h | t]}}
       end
-    end
+    end)
   end
 
   @spec clear_stack() :: :ok
   def clear_stack do
-    cast fn state ->
+    cast(fn state ->
       %{state | stack: [], post_config: []}
-    end
+    end)
   end
 
   @spec recur((() -> result)) :: result when result: var
   def recur(fun) do
-    cast fn %{stack: [h | t]} = state ->
+    cast(fn %{stack: [h | t]} = state ->
       %{state | stack: [%{h | recursing?: true} | t]}
-    end
+    end)
 
     try do
       fun.()
     after
-      cast fn %{stack: [h | t]} = state ->
+      cast(fn %{stack: [h | t]} = state ->
         %{state | stack: [%{h | recursing?: false} | t]}
-      end
+      end)
     end
   end
 
   @spec recursing :: module | nil
   def recursing do
-    get fn %{stack: stack} ->
-      Enum.find_value stack, & &1.recursing? and &1.name
-    end
+    get(fn %{stack: stack} ->
+      Enum.find_value(stack, &(&1.recursing? and &1.name))
+    end)
   end
 
   @spec read_cache(term) :: term
   def read_cache(key) do
-    get fn %{cache: cache} ->
+    get(fn %{cache: cache} ->
       cache[key]
-    end
+    end)
   end
 
   @spec write_cache(term, term) :: :ok
   def write_cache(key, value) do
-    cast fn state ->
+    cast(fn state ->
       %{state | cache: Map.put(state.cache, key, value)}
-    end
+    end)
+
     value
   end
 
   @spec clear_cache :: :ok
   def clear_cache do
-    cast fn state ->
+    cast(fn state ->
       %{state | cache: %{}}
-    end
+    end)
   end
 
   defp find_project_named(name, stack) do
-    name && Enum.find_value(stack, fn
-      %{name: n, file: file} when n === name -> file
-      %{} -> nil
-    end)
+    name &&
+      Enum.find_value(stack, fn
+        %{name: n, file: file} when n === name -> file
+        %{} -> nil
+      end)
```

Comment:
Can you please refactor this to an `if`?

```elixir
if name do
  name
else
  Enum.find_value(stack, fn
    ...
  end)
end
```

Reply:
The correct refactoring using an `if` is
```elixir
if name do
  Enum.find_value(...)
end
```
Would matching on the `name` argument be preferable?
```elixir
  defp find_project_named(nil, _stack), do: nil

  defp find_project_named(name, stack) do
    Enum.find_value(...)
  end
```